src/coreclr/jit/lsra.cpp

```diff
@@ -8208,6 +8219,10 @@ void LinearScan::addResolution(
                !blockInfo[block->bbNum].hasEHBoundaryIn);
         insertionPointString = "top";
     }
+
+    // We should never add resolution move inside BBCallAlwaysPairTail.
+    assert(!block->isBBCallAlwaysPairTail());
```

Comment:
Might be a good place to use `noway_assert`.

Reply:
Done.